lib/services/local/users.go

```diff
@@ -73,6 +73,83 @@ func (s *IdentityService) DeleteAllUsers(ctx context.Context) error {
 	return trace.Wrap(s.DeleteRange(ctx, startKey, backend.RangeEnd(startKey)))
 }
 
+// ListUsers returns the next page of users.
+func (s *IdentityService) ListUsers(ctx context.Context, pageSize int, nextToken string, withSecrets bool) ([]types.User, string, error) {
+	rangeStart := backend.Key(webPrefix, usersPrefix, nextToken)
+	rangeEnd := backend.RangeEnd(backend.ExactKey(webPrefix, usersPrefix))
+
+	// Adjust page size, so it can't be too large.
+	if pageSize <= 0 || pageSize > 1000 {
+		pageSize = 1000
+	}
+
+	// Artificially inflate the limit to account for user secrets
+	// which have the same prefix.
+	limit := pageSize * 4
+
+	var out []types.User
+	for {
+		result, err := s.Backend.GetRange(ctx, rangeStart, rangeEnd, limit)
+		if err != nil {
+			return nil, "", trace.Wrap(err)
+		}
+
+		if withSecrets {
+			collected, _, err := collectUserItems(result.Items)
+			if err != nil {
+				return nil, "", trace.Wrap(err)
+			}
+
+			for uname, uitems := range collected {
+				user, err := userFromUserItems(uname, uitems)
+				if err != nil {
+					return nil, "", trace.Wrap(err)
+				}
+				out = append(out, user)
+			}
```

Comment:
Why are we guaranteed to get the full user data from a `GetRange` that might end halfway through the items belonging to the user?